scripts/build/TestImpactAnalysis/tiaf.py

```diff
@@ -18,16 +18,227 @@
+                self._can_rerun_with_instrumentation = True
+                if self._persistent_storage.has_historic_data:
+                    logger.info("Historic data found.")
+                    self._handle_historic_data()
+                else:
+                    logger.info("No historic data found.")
+
+                # Determining our sequence type:
+                if self._has_change_list:
+                    if self._is_source_of_truth_branch:
+                        # Use TIA sequence (instrumented subset of tests) for coverage updating branches so we can update the coverage data with the generated coverage
+                        sequence_type = "tia"
+                    else:
+                        # Use TIA no-write sequence (regular subset of tests) for non coverage updating branche
+                        sequence_type = "tianowrite"
+                        # Ignore integrity failures for non coverage updating branches as our confidence in the
+                        self._runtime_args.append("--ipolicy=continue")
+                        logger.info(
+                            "Integration failure policy is set to 'continue'.")
+                    self._runtime_args.append(f"--changelist={self._change_list_path}")
+                    logger.info(f"Change list is set to '{self._change_list_path}'.")
+                else:
+                    if self._is_source_of_truth_branch and self._can_rerun_with_instrumentation:
+                        # Use seed sequence (instrumented all tests) for coverage updating branches so we can generate the coverage bed for future sequences
+                        sequence_type = "seed"
+                        # We always continue after test failures when seeding to ensure we capture the coverage for all test targets
+                        args['test_failure_policy'] = "continue"
+                    else:
+                        # Use regular sequence (regular all tests) for non coverage updating branches as we have no coverage to use nor coverage to update
+                        sequence_type = "regular"
+                        # Ignore integrity failures for non coverage updating branches as our confidence in the
+                        self._runtime_args.append("--ipolicy=continue")
+                        logger.info(
+                            "Integration failure policy is set to 'continue'.")
+
+        self._parse_arguments_to_runtime(
+            args, sequence_type, self._runtime_args)
+
+    def _parse_arguments_to_runtime(self, args, sequence_type, runtime_args):
+        """
+        Fetches the relevant keys from the provided dict, and applies the values of the arguments(or applies them as a flag) to our runtime_args list.
+
+        @param args: Dict containing the arguments passed to this TestImpact object. Will contain all the runtime arguments we need to apply.
```

Comment:
nit: Again here in the comment it's not bad to spell out 'Dictionary'

Reply:
Noted, have fixed.